linera-storage-service/src/client.rs

```diff
@@ -2,37 +2,36 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{
-    common::{ServiceContextError, ServiceStoreConfig},
+    common::{KeyTag, ServiceContextError, ServiceStoreConfig, MAX_PAYLOAD_SIZE},
     key_value_store::{
         statement::Operation, store_processor_client::StoreProcessorClient, KeyValue,
-        ReplyContainsKey, ReplyExistsNamespace, ReplyFindKeyValuesByPrefix, ReplyFindKeysByPrefix,
-        ReplyListAll, ReplyReadMultiValues, ReplyReadValue, RequestClearJournal,
-        RequestContainsKey, RequestCreateNamespace, RequestDeleteAll, RequestDeleteNamespace,
-        RequestExistsNamespace, RequestFindKeyValuesByPrefix, RequestFindKeysByPrefix,
-        RequestListAll, RequestReadMultiValues, RequestReadValue, RequestWriteBatch, Statement,
+        KeyValueAppend, ReplyContainsKey, ReplyExistsNamespace, ReplyFindKeyValuesByPrefix,
+        ReplyFindKeysByPrefix, ReplyListAll, ReplyReadMultiValues, ReplyReadValue,
+        ReplySpecificChunk, RequestContainsKey, RequestCreateNamespace, RequestDeleteAll,
+        RequestDeleteNamespace, RequestExistsNamespace, RequestFindKeyValuesByPrefix,
+        RequestFindKeysByPrefix, RequestListAll, RequestReadMultiValues, RequestReadValue,
+        RequestSpecificChunk, RequestWriteBatchExtended, Statement,
     },
 };
-use async_lock::{RwLock, Semaphore, SemaphoreGuard};
+use async_lock::{RwLock, RwLockWriteGuard, Semaphore, SemaphoreGuard};
 use async_trait::async_trait;
+use linera_base::ensure;
 use linera_views::{
-    batch::Batch,
+    batch::{Batch, WriteOperation},
     common::{
         AdminKeyValueStore, CommonStoreConfig, KeyValueStore, ReadableKeyValueStore,
-        WritableKeyValueStore, MIN_VIEW_TAG,
+        WritableKeyValueStore,
     },
 };
-use std::sync::Arc;
+use serde::de::DeserializeOwned;
+use std::{mem, sync::Arc};
 use tonic::transport::{Channel, Endpoint};
 
 #[cfg(any(test, feature = "test"))]
 use linera_views::test_utils::generate_test_namespace;
 
-/// Key tags to create the sub keys used for storing data on storage.
-#[repr(u8)]
-pub(crate) enum KeyTag {
-    /// Prefix for the storage of the keys of the map
-    Key = MIN_VIEW_TAG,
-}
+// The maximum key size is set to 1M rather arbitrarily.
```

Comment:
I guess what matters is that it's between the maximum key size we use in Linera, and the maximum size of a gRPC message payload?